db-engine/src/main/java/com/splicemachine/db/impl/sql/catalog/SYSTABLESRowFactory.java

```diff
@@ -52,661 +52,664 @@
+                                null, null, view, viewId, 0, 0, 0),
+                        new ColumnDescriptor(SCHEMAID, 4, 4,
+                                DataTypeDescriptor.getBuiltInDataTypeDescriptor(Types.CHAR, false, 36),
+                                null, null, view, viewId, 0, 0, 0),
+                        new ColumnDescriptor(LOCKGRANULARITY, 5, 5,
+                                DataTypeDescriptor.getBuiltInDataTypeDescriptor(Types.CHAR, false, 1),
+                                null, null, view, viewId, 0, 0, 0),
+                        new ColumnDescriptor(VERSION, 6, 6,
+                                DataTypeDescriptor.getBuiltInDataTypeDescriptor(Types.VARCHAR, false, 128),
+                                null, null, view, viewId, 0, 0, 0),
+                        new ColumnDescriptor(COLSEQUENCE, 7, 7,
+                                DataTypeDescriptor.getBuiltInDataTypeDescriptor(Types.INTEGER, false),
+                                null, null, view, viewId, 0, 0, 0),
+                        new ColumnDescriptor(DELIMITED, 8, 8,
+                                DataTypeDescriptor.getBuiltInDataTypeDescriptor(Types.VARCHAR, false),
+                                null, null, view, viewId, 0, 0, 0),
+                        new ColumnDescriptor(ESCAPED, 9, 9,
+                                DataTypeDescriptor.getBuiltInDataTypeDescriptor(Types.VARCHAR, false),
+                                null, null, view, viewId, 0, 0, 0),
+                        new ColumnDescriptor(LINES, 10, 10,
+                                DataTypeDescriptor.getBuiltInDataTypeDescriptor(Types.VARCHAR, true),
+                                null, null, view, viewId, 0, 0, 0),
+                        new ColumnDescriptor(STORED, 11, 11,
+                                DataTypeDescriptor.getBuiltInDataTypeDescriptor(Types.VARCHAR, true),
+                                null, null, view, viewId, 0, 0, 0),
+                        new ColumnDescriptor(LOCATION, 12, 12,
+                                DataTypeDescriptor.getBuiltInDataTypeDescriptor(Types.VARCHAR, true),
+                                null, null, view, viewId, 0, 0, 0),
+                        new ColumnDescriptor(COMPRESSION, 13, 13,
+                                DataTypeDescriptor.getBuiltInDataTypeDescriptor(Types.VARCHAR, true),
+                                null, null, view, viewId, 0, 0, 0),
+                        new ColumnDescriptor(IS_PINNED, 14, 14,
+                                DataTypeDescriptor.getBuiltInDataTypeDescriptor(Types.BOOLEAN, false),
+                                null, null, view, viewId, 0, 0, 0),
+                        new ColumnDescriptor(PURGE_DELETED_ROWS, 15, 15,
+                                DataTypeDescriptor.getBuiltInDataTypeDescriptor(Types.BOOLEAN, false),
+                                null, null, view, viewId, 0, 0, 0),
+                        new ColumnDescriptor(MIN_RETENTION_PERIOD, 16, 16,
```

Comment:
As @jyuanca mentioned, we need an upgrade script to add the new column to the table for existing system. The upgrade script should also cover the update of the view. Please refer to SpliceDataDictionary.updateColumnViewInSys() for the logic to update the sysvw.systablesview.

Reply:
Thanks, I addressed your comment, but not sure if it is exactly correctly.